### PR TITLE
✨ Update `ORJSONResponse` to support non `str` keys and serializing Numpy arrays

### PR DESCRIPTION
**Pull request by baby5**
the detail see: https://github.com/ijl/orjson/issues/197

my case is `<class 'sqlalchemy.sql.elements.quoted_name'>`

### CODE REVIEW

**Review by adam-tokarski — CHANGES_REQUESTED** [review]
Please provide any tests for that.

**Comment by baby5** [comment]
@adam-tokarski done

File: tests/test_default_response_class.py [hunk 1]
```diff
@@ -31,6 +32,12 @@ def get_root():
     return {"msg": "Hello World"}
 
 
+@app.get("/orjson_non_str_keys")
+def get_orjson_non_str_keys():
+    key = quoted_name("msg")
+    return {key: "Hello World"}
```
Comment (adam-tokarski) [inline tests/test_default_response_class.py:38]: Isn't that `key` still a string? It seems like it should be working even before the fix (I didn't check that though).

Also, you could provide a test with some integer as a key maybe?

**Review by adam-tokarski — APPROVED** [review]

**Comment by tiangolo** [comment]
Great, thank you @baby5! And thanks for adding tests! :cake: 

And thanks @adam-tokarski for the help and review! :coffee:

**Comment by NikosAlexandris** [comment]
What is the status of this issue regarding support for NumPy arrays ?

**Pull request merged**